massa-bootstrap/src/client.rs

```diff
@@ -371,16 +372,14 @@ async fn send_client_message(
 }
 
 async fn connect_to_server(
-    establisher: &mut Establisher,
+    establisher: &mut impl BSEstablisher,
```

Comment:
Monomorphisation will cause this function to be compiled once for each impl. In non-test builds, It's only the default establisher.

If we want to make many different establishers, we can use this pattern:
```rust
fn foo(many_impl: impl ToString) {
    fn concrete_foo(the_string) {
        // do the string thing
    }
    concrete_foo(many_impl.to_string())
```

This is an ugly pattern, and is one of rusts flaws that this optimisation needs a manual opt-in, but is available if needed (can explain whats happening if needed)